x-pack/plugins/maps/public/components/layer_addpanel/view.js

```diff
@@ -23,21 +23,37 @@ import {
 
 export class AddLayerPanel extends Component {
 
-  state = {
-    sourceType: null,
+  constructor() {
+    super();
+
+    this.state = {
```

Comment:
no need to change the idiomatic 

`state = {
 sourceType: null
} `

Retain that form. Then we also don't need a new constructor.